api/clients/v2/examples/client_construction.go

```diff
@@ -363,3 +438,153 @@ func createLogger() (logging.Logger, error) {
 
 	return logger, nil
 }
+
+func createContractDirectory(
+	ctx context.Context,
+	logger logging.Logger,
+	ethClient common.EthClient,
+) (*directory.ContractDirectory, error) {
+	directoryAddress := gethcommon.HexToAddress(contractRegistryAddress)
+	contractDirectory, err := directory.NewContractDirectory(ctx, logger, ethClient, directoryAddress)
+	if err != nil {
+		return nil, fmt.Errorf("new contract directory: %w", err)
+	}
+	return contractDirectory, nil
+}
+
+func createClientLedger(
+	ctx context.Context,
+	logger logging.Logger,
+	mode clientledger.ClientLedgerMode,
+	ethClient common.EthClient,
+	accountID gethcommon.Address,
+	contractDirectory *directory.ContractDirectory,
+	disperserClient *clients.DisperserClient,
+) (*clientledger.ClientLedger, error) {
+	paymentVaultAddr, err := contractDirectory.GetContractAddress(ctx, directory.PaymentVault)
+	if err != nil {
+		return nil, fmt.Errorf("get PaymentVault address: %w", err)
+	}
+
+	paymentVault, err := vault.NewPaymentVault(logger, ethClient, paymentVaultAddr)
+	if err != nil {
+		return nil, fmt.Errorf("new payment vault: %w", err)
+	}
+
+	minNumSymbols, err := paymentVault.GetMinNumSymbols(ctx)
```

Comment:
does the client monitor for changes to this?
Also for my own knowledge who owns the source of truth wrt this? is it validators or dispersers? afaiu this is only used for disperser charging client, which the disperser doesn't really need to follow the onchain logic right? So it could set/use whatever, and following the contract is just a convention?